### PR TITLE
Make okhttp3 ,feign http9.x plugin to support java 14 

### LINKED ISSUE #5728: RealCallInterceptor not support java14
RealCallInterceptor  

line 73

 Field headersField = Request.class.getDeclaredField("headers");
 Field modifiersField = Field.class.getDeclaredField("modifiers");
modifiers can not access in java14

image

### LINKED ISSUE #5730: Agent plugin for Kafka 2.1.0+
kafka client 2.3.1

image

### PR DESCRIPTION
**Pull request by hsoftxl**
### okhttp3 ,feign http9.x plugin support java 14

- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking/blob/master/docs/en/guides/Plugin-test.md) **NO NEED**
- [x] Add a component id in [the component-libraries.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml). **NO NEED**
- [x] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets). **NO NEED**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #5728.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/c2141978d1039375598a32418b75161a78da22c1/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please rebase your comments and remove Chinese

File: CHANGES.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ Release Notes.
 #### Project
 
 #### Java Agent
+plugin feign-default-http-9.x-plugin support java 14
```
Comment (wu-sheng) [inline CHANGES.md:10]:
```suggestion
Make thr feign plugin to support java 14
```

File: CHANGES.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ Release Notes.
...
+plugin okhttp-3.x-plugin support java 14
```
Comment (wu-sheng) [inline CHANGES.md:11]: Same style as above

**Comment by wu-sheng** [comment]
Seems you are closing wrong issue?

File: CHANGES.md (hunk 2 above)
Comment (hsoftxl) [reply CHANGES.md]: modified

File: apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java [hunk 3]
```diff
@@ -70,10 +69,6 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         SpanLayer.asHttp(span);
 
         Field headersField = Request.class.getDeclaredField("headers");
-        Field modifiersField = Field.class.getDeclaredField("modifiers");
-        modifiersField.setAccessible(true);
-        modifiersField.setInt(headersField, headersField.getModifiers() & ~Modifier.FINAL);
-
         headersField.setAccessible(true);
         Headers.Builder headerBuilder = request.headers().newBuilder();
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java:73]: Note, the existing code creates a new builder, but never read the existing headers, if the client has meaning headers, such as token, they would be lost, right?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java:73]: I know this is not related to this fix. You could consider to check these 3 plugins, to submit issue or another PR to fox this.

**Review by wu-sheng — APPROVED** [review]

File: apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java (hunk 3 above)
Comment (hsoftxl) [reply apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java:73]: first question : request.headers().newBuilder(); 
this code read existing headers, and next step put new values in it.

2: there just 3 block code use this way to change modifiers  in project.

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java:73]: Thanks for clearance. Good to me.

**Comment by wu-sheng** [comment]
@dmsolr the finagle test fails even I have retried many times. It makes the CI wouldn't check the feign plugin which relates to this.

**Comment by kezhenxu94** [comment]
> @dmsolr the finagle test fails even I have retried many times. It makes the CI wouldn't check the feign plugin which relates to this.

I notice that some dependencies disappear in the repo, checking why now

**Comment by wu-sheng** [comment]
He pinged me and told me the same thing. Consider to remove that version in test.

**Comment by dmsolr** [comment]
> I notice that some dependencies disappear in the maven repo, checking why now

I had checked the maven repo that it is certainly not found.
So I am going to remove this scenario.

**Review by kezhenxu94 — APPROVED** [review]
Reapprove

**Pull request merged**